test/module/irohad/consensus/yac/peer_orderer_test.cpp

```diff
@@ -82,3 +83,36 @@ TEST_F(YacPeerOrdererTest, PeerOrdererOrderingWhenInvokeFaillCase) {
   auto order = orderer.getOrdering(YacHash());
   ASSERT_EQ(order, nonstd::nullopt);
 }
+
+TEST_F(YacPeerOrdererTest, FairnessTest) {
+  // Calculate number of permutations of peers
+  size_t comb = std::tgamma(N_PEERS+1);
+  // Run experiments 30 times for each combination
+  size_t times = comb * 30;
+  std::unordered_map<std::string, int> histogram;
+  EXPECT_CALL(*wsv, getLedgerPeers())
+      .Times(times)
+      .WillRepeatedly(Return(peers));
+  for (size_t t = 1; t <= times; ++t) {
+    std::string hash_string =   std::to_string(t);
+    auto order = orderer.getOrdering(YacHash(hash_string, hash_string));
+    auto current_peers = order.value().getPeers();
+    std::string res = std::accumulate(current_peers.begin(),
+                                      current_peers.end(),
+                                      std::string(),
+                                      [](std::string res, const auto &peer) {
+                                        return res + " " + peer.address;
+                                      });
+    histogram[res]++;
+  }
+  ASSERT_EQ(comb, histogram.size());
+
+  // Output historgram, should output uniform distribution
+  std::cout << "The histogram output: " << std::endl;
+  std::for_each(histogram.begin(),
```

Comment:
Maybe it's better: to check whether `accumulate(..)/times` in some suburb of 1? Otherwise this test is pretty pointless

Reply:
But in this case will be potential fails of the test because of "bad" random.